src/index.js

```diff
@@ -245,6 +248,12 @@ async function run(params) {
     throw new Error('ref parameter missing.');
   }
 
+  const {
+    __OW_ACTION_NAME: actionName,
+  } = process.env;
+
+  const pkgName = actionName ? actionName.match(/^\/([^/]+\/)+[^/]+$/)[1] : '';
```

Comment:
In a fully qualified action name such as `/helix/helix-observation/index-files@2.2.1`, the group 1 will contain `helix-observation/`.

Reply:
are you sure that `__OW_ACTION_NAME` is fully qualified?

Reply:
I am: https://github.com/apache/openwhisk/blob/master/docs/actions.md

Reply:
but only, if the action resides in a package :-) otherwise it's just `/namespace/actionname`....
btw: I find `actionName.split('/')[2]` much more readable.